src/services/nifSemanticValidationService.js

```diff
@@ -0,0 +1,100 @@
+services.factory('nifSemanticValidationService', [nifSemanticValidationService]); 
+
+function nifSemanticValidationService() {
+	
+	function isDNISemanticallyCorrect(dni) {
+		return isDNIControlDigitCorrect(dni);
+	}
+
+	function isNIESemanticallyCorrect(nie) {
```

Comment:
if you assume NIE must be well formed before calling this method, add a comment on top indicating it.